### ISSUE TITLE
QLDB Driver V3 timeout

### ISSUE THREAD

**Issue by mkhraisha** [awslabs/amazon-qldb-driver-nodejs#233]
Hi,

We're running into an issue when we upgraded to the QLDB V3 SDK (v3.0.0) where the driver times out after a period of inactivity ~roughly 17-20 min, this was not an issue in v2.2.

we have configured the agent as follows (directly from the changelog docs): 
```
import { Agent } from 'https';
import { QldbDriver, RetryConfig  } from 'amazon-qldb-driver-nodejs';

const maxConcurrentTransactions: number = 10;

const agentForQldb: Agent = new Agent({
    keepAlive: true,
    maxSockets: maxConcurrentTransactions
});

const serviceConfigurationOptions = {
    region: "us-east-1",
    httpOptions: {
        agent: agentForQldb
    }
};

const qldbDriver: QldbDriver = new QldbDriver("testLedger", serviceConfigurationOptions, maxConcurrentTransactions);
```

To reproduce issue, instantiate driver, call a transaction, wait 20 minutes, then call another transaction on the driver. 
When calling execute lambda on the same driver we end up with ``` Cannot read properties of null (reading 'executeLambda') ```.

What is the correct pattern here? the docs mention that the driver is responsible for all session management, so was under the impression it should be able to handle delayed calls.

**Comment by mato-gallardo** [comment]
I've checked the code and we have this:
`for (let retryAttempt: number = 1; true; retryAttempt++) {
            let session: QldbSession = null;
            try {
                if (replaceDeadSession) {
                    await createNewSession(this);
                } else {
                    session = await getSession(this);
                }
                return await session.executeLambda(transactionLambda);
            } catch (e) {`

and it should be `session = await createNewSession(this);` when the session is dead.
I've tried the connectionTimeout but when load testing it continues to give the error.

**Comment by mato-gallardo** [comment]
I have a PR request for fixing this.  https://github.com/awslabs/amazon-qldb-driver-nodejs/pull/243

**Comment by bwinchester** [comment]
Thanks @mato-gallardo for the PR. It has been merged and a v3.0.1 patch has been published here and npm.
